app.ts

```diff
@@ -0,0 +1,153 @@
+﻿let firstIndex:number, lastIndex:number; // stores the indices of the first and last rows in the table
+let recordCount:number; // stores the number of records in the database
+let resizeTimer; // stores time since last window resize
+
+// builds the grid table from the column headers and row data
+function buildTable(columns, records) {
+
+	// create table
+	let table = document.createElement("table");
+	table.className = "gridtable";
+    let thead = document.createElement("thead");
+    let tbody = document.createElement("tbody");
+	let headRow = document.createElement("tr");
+
+	// create column headers
+    for(const header of columns) {
+		var th = document.createElement("th");
+		th.appendChild(document.createTextNode(header));
+		headRow.appendChild(th);
+    };
+    thead.appendChild(headRow);
+	table.appendChild(thead);
+
+	// create rows
+    records.forEach(function (el) {
+		let tr = document.createElement("tr");
+		for (var o in el) {
+			let td = document.createElement("td");
+			td.appendChild(document.createTextNode(el[o]))
+			tr.appendChild(td);
+		}
+		tbody.appendChild(tr);
+	});
+	table.appendChild(tbody);
+
+	// builds footer elements (buttons, search field)
+	document.getElementById("footer").innerHTML = `<a onclick="goPrevious()">&laquo;</a>
+	<a onclick="goNext()">&raquo;</a>
+	<input type="text" id="search" onkeyup="search(event)" placeholder="Enter ID...">`;
+
+	document.getElementById("loader").remove();
+
+    return table;
+}
+
+// filters rows by user input ID
+function search(e){
+	let input = $("#search").val().toString();
+	let $table = $("tbody tr").toArray();
+
+	for(const record of $table){
+		let td = record.getElementsByTagName("td")[0];
+		record.style.display="block";
+		if(td){
+			let val = td.innerText || td.textContent;
+			if(val.indexOf(input) > -1){
+				record.style.display = "";
+			}
+			else{
+				record.style.display = "none";
+			} 		
+		}
+	}
+
+	// if record doesn't exist on current page, query database and display results
+	if(e.which == 13)
+	{
+		let end = parseInt(input) + calculateRows();
+		clearTable();
```

Comment:
Since you always `clearTable` before you `loadPage` you can move `clearPage` inside `loadPage`